### PR TITLE
[GIT] Ignore warning, error message

### PR DESCRIPTION
**Pull request by seonggwonyoon**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Bug fix

**What is the current behavior? (You can also link to an open issue here)**
https://github.com/withfig/fig/issues/107

**What is the new behavior (if this is a feature change)?**

**Additional info:**

### CODE REVIEW

**Comment by QuiiBz** [comment]
Hi, thanks a lot for your first contribution! Based on Matt's comment (https://github.com/withfig/fig/issues/107#issuecomment-852730125), shouldn't it filter out anything starting with `error` too, not only `warning`?

**Comment by seonggwonyoon** [comment]
> Hi, thanks a lot for your first contribution! Based on Matt's comment ([withfig/fig#107 (comment)](https://github.com/withfig/fig/issues/107#issuecomment-852730125)), shouldn't it filter out anything starting with `error` too, not only `warning`?

Thank you @QuiiBz 
I missed `error` option! just added.

**Review by QuiiBz — COMMENTED** [review]
I think we can simplify this and create a top-level function to avoid duplication. Something like this (not tested):

```ts
const filter = (out: string): string => {
  return (out.startsWith("warning:") || out.startsWith("error:"))
    ? out = out.split("\n").slice(1).join("\n")
    : out;
}
```

So we will only call `filter` this way:

```ts
out = filter(out);
```

**Comment by seonggwonyoon** [comment]
> I think we can simplify this and create a top-level function to avoid duplication. Something like this (not tested):
> 
> ```ts
> const filter = (out: string): string => {
>   return (out.startsWith("warning:") || out.startsWith("error:"))
>     ? out.split("\n").slice(1).join("\n")
>     : out;
> }
> ```
> 
> So we will only call `filter` this way:
> 
> ```ts
> out = filter(out);
> ```

As you said, I applied the filter function!

**Review by QuiiBz — APPROVED** [review]
That's cleaner!

File: dev/git.ts [hunk 1]
```diff
@@ -1,8 +1,16 @@
+const filter = (out: string): string => {
```
Comment (cstrnt) [inline dev/git.ts:1]: I honestly would prefer being more verbose here: Maybe `filterWarnings`? I think functions having the same name as builtins might confuse other contributors

**Review by cstrnt — CHANGES_REQUESTED** [review]

File: dev/git.ts (hunk 1 above)
Comment (seonggwonyoon) [reply dev/git.ts]: Fix `filter` to `filterWarnings`!

File: dev/git.ts [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+    ? out.split("\n").slice(1).join("\n")
+    : out;
+};
+
 const gitGenerators: Record<string, Fig.Generator> = {
   // Commit history
   commits: {
     script: "git log --oneline",
     postProcess: function (out) {
+      out = filterWarnings(out);
```
Comment (cstrnt) [inline dev/git.ts:12]: Could you please create a const here. Re-assigning to a parameter is a code smell imho

File: dev/git.ts [hunk 3]
```diff
@@ -45,6 +55,8 @@ const gitGenerators: Record<string, Fig.Generator> = {
   treeish: {
     script: "git diff --cached --name-only",
     postProcess: function (out) {
+      out = filterWarnings(out);
```
Comment (cstrnt) [inline dev/git.ts:58]: see above

File: dev/git.ts [hunk 4]
```diff
@@ -64,9 +76,12 @@ const gitGenerators: Record<string, Fig.Generator> = {
   branches: {
     script: "git branch --no-color",
     postProcess: function (out) {
+      out = filterWarnings(out);
```
Comment (cstrnt) [inline dev/git.ts:79]: see above

File: dev/git.ts [hunk 5]
```diff
@@ -132,9 +147,12 @@ const gitGenerators: Record<string, Fig.Generator> = {
   files_for_staging: {
     script: "git status --short",
     postProcess: (output, context) => {
+      output = filterWarnings(output);
```
Comment (cstrnt) [inline dev/git.ts:150]: see above

**Review by cstrnt — CHANGES_REQUESTED** [review]

File: dev/git.ts (hunk 2 above)
Comment (QuiiBz) [reply dev/git.ts:12]: Btw, this function’s name not only filters warnings but also errors. Shouldn’t we rename it to something more generic?

Comment (seonggwonyoon) [reply dev/git.ts:12]: @QuiiBz Yes, it filters out warnings and errors.
Let's think about changing it to a more generic name.

Comment (seonggwonyoon) [reply dev/git.ts:12]: @cstrnt I agree with the saying Re-assigning to a parameter is a code smell.
But I don't understand the phrase `create a const here' Can you explain a bit more?

Comment (seonggwonyoon) [reply dev/git.ts:12]: @cstrnt You mean this code?
```typescript
postProcess: function (out) {
      const ?? = filterWarnings(out);
      if (??.startsWith("fatal:")) {
        return [];
      }
```

Comment (cstrnt) [reply dev/git.ts:12]: hey @seonggwonyoon, sorry for not replying :/ That's exactly what I mean :)

Comment (seonggwonyoon) [reply dev/git.ts]: @cstrnt Create const is done!
@QuiiBz Fix filterWarnings to filterMessages

**Review by QuiiBz — APPROVED** [review]
LGTM

**Comment by cstrnt** [comment]
Hey sorry to be annoying, but could you please run `npm run lint:fix` in order to to fix the styling :) After this is done, we will merge the PR 😊

**Comment by seonggwonyoon** [comment]
> Hey sorry to be annoying, but could you please run `npm run lint:fix` in order to to fix the styling :) After this is done, we will merge the PR 😊

Fixed it. Thanks @cstrnt

**Pull request merged**